hack/jenkins/e2e.sh

```diff
@@ -190,6 +204,194 @@ case ${JOB_NAME} in
+    : ${CMD_GROUP:="beta"}
+    : ${DOGFOOD_GCLOUD:="true"}
+    : ${E2E_CLUSTER_NAME:="jkns-gke-e2e-ci"}
+    : ${E2E_NETWORK:="e2e-gke-ci"}
+    : ${E2E_SET_CLUSTER_API_VERSION:="y"}
+    : ${GINKGO_TEST_ARGS:="--ginkgo.skip=${GCE_DEFAULT_SKIP_TEST_REGEX}|${GCE_FLAKY_TEST_REGEX}"}
+    : ${PROJECT:="k8s-jkns-e2e-gke-ci"}
+    ;;
+
+  # Launch a GKE cluster at latest CI, run the reboot tests and the non-flaky
+  # tests sequentially.
+  kubernetes-e2e-gke-ci-reboot)
+    : ${CMD_GROUP:="beta"}
+    : ${DOGFOOD_GCLOUD:="true"}
+    : ${E2E_CLUSTER_NAME:="jenkins-gke-e2e-ci-reboot"}
+    : ${E2E_NETWORK:="e2e-gke-ci"}
+    : ${E2E_SET_CLUSTER_API_VERSION:="y"}
+    : ${GINKGO_TEST_ARGS:="--ginkgo.skip=${GCE_SKIP_KEEP_REBOOT_TEST_REGEX}|${GCE_FLAKY_TEST_REGEX}"}
+    : ${PROJECT:="k8s-jkns-e2e-gke-ci"}
+    ;;
+
+  # Launch a GKE cluster in test, run non-flaky tests sequentially.
+  kubernetes-e2e-gke-test)
+    : ${CMD_GROUP:="beta"}
+    : ${DOGFOOD_GCLOUD:="true"}
+    : ${E2E_CLUSTER_NAME:="jkns-gke-e2e-test"}
+    : ${E2E_NETWORK:="e2e-gke-test"}
+    : ${GINKGO_TEST_ARGS:="--ginkgo.skip=${GCE_DEFAULT_SKIP_TEST_REGEX}|${GCE_FLAKY_TEST_REGEX}"}
+    : ${JENKINS_USE_RELEASE_TARS:="y"}
+    # TODO(mbforbes): Do we need to set PATH here?
+    : ${PATH="/usr/local/share/google/google-cloud-sdk/bin:${PATH}"}
+    : ${PROJECT:="k8s-jkns-e2e-gke-ci"}
+    # TODO(mbforbes): Do we need to set CLOUDSDK_CORE_PROJECT here?
+    : ${CLOUDSDK_CORE_PROJECT:="${PROJECT}"}
+    ;;
+
+  # Launch a GKE cluster in staging, run non-flaky tests sequentially.
+  kubernetes-e2e-gke-staging)
+    : ${CMD_GROUP:="beta"}
+    : ${DOGFOOD_GCLOUD:="true"}
+    : ${E2E_CLUSTER_NAME:="jenkins-gke-e2e-staging"}
+    : ${E2E_NETWORK:="e2e-gke-staging"}
+    # TODO(mbforbes): Do we need to set E2E_SET_CLUSTER_API_VERSION here?
+    : ${E2E_SET_CLUSTER_API_VERSION:="y"}
+    : ${GINKGO_TEST_ARGS:="--ginkgo.skip=${GCE_DEFAULT_SKIP_TEST_REGEX}|${GCE_FLAKY_TEST_REGEX}"}
+    : ${JENKINS_EXPLICIT_VERSION:="release/v0.21.0"}
+    # TODO(mbforbes): Do we need to set PATH here?
+    : ${PATH="/usr/local/share/google/google-cloud-sdk/bin:${PATH}"}
+    : ${PROJECT:="k8s-jkns-e2e-gke-ci"}
+    # TODO(mbforbes): Do we need to set CLOUDSDK_CORE_PROJECT here?
+    : ${CLOUDSDK_CORE_PROJECT:="${PROJECT}"}
+    ;;
+
+  # Launch a GKE cluster in prod, run non-flaky tests sequentially.
+  kubernetes-e2e-gke-prod)
+    : ${CMD_GROUP:="beta"}
+    : ${DOGFOOD_GCLOUD:="true"}
+    : ${E2E_CLUSTER_NAME:="jkns-gke-e2e-prod"}
+    : ${E2E_NETWORK:="e2e-gke-prod"}
+    : ${GINKGO_TEST_ARGS:="--ginkgo.skip=${GCE_DEFAULT_SKIP_TEST_REGEX}|${GCE_FLAKY_TEST_REGEX}"}
+    : ${JENKINS_EXPLICIT_VERSION:="release/v0.19.3"}
+    # TODO(mbforbes): Do we need to set PATH here?
+    : ${PATH="/usr/local/share/google/google-cloud-sdk/bin:${PATH}"}
+    : ${PROJECT:="k8s-jkns-e2e-gke-ci"}
+    # TODO(mbforbes): Do we need to set CLOUDSDK_CORE_PROJECT here?
+    : ${CLOUDSDK_CORE_PROJECT:="${PROJECT}"}
```

Comment:
Not needed , provided the tests run all gcloud commands with an explicit `--project=$PROJECT`. Setting `CLOUD_SDK_CORE_PROJECT` is the envvar equivalent of
```
gcloud config set project $PROJECT
```